graylog2-server/src/main/java/org/graylog2/telemetry/rest/TelemetryResource.java

```diff
@@ -0,0 +1,134 @@
+import org.graylog2.shared.rest.resources.ProxiedResource;
+import org.graylog2.shared.rest.resources.system.RemoteSystemResource;
+
+import javax.inject.Named;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+
+import static org.graylog2.audit.AuditEventTypes.TELEMETRY_USER_SETTINGS_UPDATE;
+import static org.graylog2.shared.rest.documentation.generator.Generator.CLOUD_VISIBLE;
+
+@RequiresAuthentication
+@Api(value = "Telemetry", description = "Message inputs", tags = {CLOUD_VISIBLE})
+@Path("/telemetry")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class TelemetryResource extends ProxiedResource {
+
+    private final TelemetryService telemetryService;
+    private final AuditEventSender auditEventSender;
+
+    protected TelemetryResource(NodeService nodeService,
+                                RemoteInterfaceProvider remoteInterfaceProvider,
+                                @Context HttpHeaders httpHeaders,
+                                @Named("proxiedRequestsExecutorService") ExecutorService executorService,
+                                TelemetryService telemetryService,
+                                AuditEventSender auditEventSender) {
+        super(httpHeaders, nodeService, remoteInterfaceProvider, executorService);
+        this.telemetryService = telemetryService;
+        this.auditEventSender = auditEventSender;
+    }
+
+    @GET
+    @ApiOperation(value = "Get telemetry information.")
+    public Map<String, Object> get() {
+        return telemetryService.getTelemetryResponse(getCurrentUserOrThrow(), getSystemOverviewResponses());
+    }
+
+    @GET
+    @Path("user/settings")
+    @ApiOperation("Retrieve a user's telemetry settings.")
+    @ApiResponses({
+            @ApiResponse(code = 404, message = "Current user not found.")
+    })
+    public TelemetryUserSettings getTelemetryUserSettings() {
+        return telemetryService.getTelemetryUserSettings(getCurrentUserOrThrow());
+    }
+
+    @PUT
+    @Path("user/settings")
+    @ApiOperation("Update a user's telemetry settings.")
+    @ApiResponses({@ApiResponse(code = 404, message = "Current user not found.")})
+    @NoAuditEvent("Audit event is sent manually.")
+    public void saveTelemetryUserSettings(@ApiParam(name = "JSON body", value = "The telemetry settings to assign to the user.", required = true)
+                                          @Valid @NotNull TelemetryUserSettings telemetryUserSettings) {
+
+        User currentUser = getCurrentUserOrThrow();
```

Comment:
This is the endpoint which is also called when editing another user's settings, so either this needs to be changed to accept a user id or disabled in the UI.
Right now, changing the value for any user in the UI will result in changing my own setting.

Reply:
Oh yeah, same for `getTelemetrySettings`

Reply:
Good catch. I spoke to Ousmane, we are adjusting the UI so that a user can only change his own telemetry settings.

Reply:
Hi @moesterheld, now we are not allowing a user to change the setting for other. You can only change your own if your profile is not read-only.

Reply:
Thank you @moesterheld . I fixed the view part